tests/test_warm_reboot.py

```diff
@@ -780,3 +789,665 @@ def test_swss_port_state_syncup(dvs, testlog):
+                                      format(table + ".*" + entry + ".*" + "DEL")])
+
+    print exitcode
+    print num
+
+    assert num.strip() == "1"
+
+
+#
+# Verifies that a single change matching the passed parameters has been seen in
+# sairedis logs.
+#
+# Input parameters (example):
+#
+#       table: ROUTE
+#       entry: 192.168.200.0/24
+#       action: SET, CREATE, DELETE
+#
+# The above example would match this sairedis log entry:
+#
+# .* |s|SAI_OBJECT_TYPE_ROUTE_ENTRY:{"dest":"192.168.1.0/30" .*
+#
+def check_sairedis_change(dvs, table, entry, action):
+
+    if action == 'SET':
+        (exitcode, num) = dvs.runcmd(['sh', '-c', "egrep \"{}\" /var/log/swss/sairedis.rec | wc -l".
+                                      format("s\|" + ".*" + table + ".*" + entry)])
+
+    elif action == 'CREATE':
+        (exitcode, num) = dvs.runcmd(['sh', '-c', "egrep \"{}\" /var/log/swss/sairedis.rec | wc -l".
+                                      format("c\|" + ".*" + table + ".*" + entry)])
+
+    elif action == 'DELETE':
+        (exitcode, num) = dvs.runcmd(['sh', '-c', "egrep \"{}\" /var/log/swss/sairedis.rec | wc -l".
+                                      format("r\|" + ".*" + table + ".*" + entry)])
+
+    assert num.strip() == "1"
+
+
+def obtain_syslog_changes(dvs):
+
+    (exitcode, num) = dvs.runcmd(['sh', '-c',
+                                  'egrep "reconciliation:" /var/log/syslog | grep -v "no changes" | wc -l'])
+    return num.strip()
+
+
+#
+# Obtain the total number of changes observed in swss.rec for a particular table.
+#
+def obtain_swss_changes(dvs, table):
+    (exitcode, num) = dvs.runcmd(['sh', '-c', "egrep \"{}\" /var/log/swss/swss.rec | wc -l".
+                                  format(table + "_TABLE")])
+    return num.strip()
+
+
+#
+# Obtain the total number of changes observed in sairedis.rec for a particular table.
+#
+def obtain_sairedis_changes(dvs, table):
+    (exitcode, num) = dvs.runcmd(['sh', '-c', "egrep \"{}\" /var/log/swss/sairedis.rec | wc -l".
+                                  format("TYPE_" + table + "_ENTRY")])
+    return num.strip()
+
+
+def check_restart_timer(dvs, timer_value):
+    (exitcode, num) = dvs.runcmd(['sh', '-c',
+                                  'grep "Getting warmStartTimer" /var/log/syslog | rev | cut -d ' ' -f 1 | rev'])
+    assert num.strip() == timer_value
+
+
+# Temporary instruction to activate warm_restart. To be deleted once equivalent CLI
+# function is pushed to sonic-utils.
+def enable_warmrestart(dvs, db, app_name):
+    create_entry_tbl(
+        db,
+        swsscommon.CFG_WARM_RESTART_TABLE_NAME, app_name,
+        [
+            ("enable", "true"),
+        ]
+    )
+
+
+################################################################################
+#
+# Routing warm-restart testcases
+#
+################################################################################
+
+def test_routing_WarmRestart(dvs):
```

Comment:
>tart(dvs): [](start = 24, length = 10)

add (testlog) param, it will print out separator for this test in the syslog.

Reply:
done